File: yourlibrary/lib/src/models/author_model.dart
// To parse this JSON data, do
//
//     final author = authorFromJson(jsonString);

import 'dart:convert';

Author authorFromJson(String str) => Author.fromJson(json.decode(str));

String authorToJson(Author data) => json.encode(data.toJson());

class Author {
  Author({
    this.names,
    this.surnames,
    this.placeBorn,
    this.dateBorn,
  });

  String? names;
  String? surnames;
  String? placeBorn;
  String? dateBorn;

  factory Author.fromJson(Map<String, dynamic> json) => Author(
        names: json["names"],
        surnames: json["surnames"],
        placeBorn: json["place_born"],
        dateBorn: json["date_born"],
      );

  Map<String, dynamic> toJson() => {
        "names": names,
        "surnames": surnames,
        "place_born": placeBorn,
        "date_born": dateBorn,
      };
}

File: backend-yourlibrary/functions/src/models/authorModel.ts
export interface Author {
    idauthor?: string,
    names: string,
    surnames: string,
    placeBorn: string,
    dateBorn: string,
    idbook: string
}

export function Author(data: any, id?:string) {
    const { names, surnames, placeBorn, dateBorn, idbook } = data;

    let object: Author = {
        idauthor: id,
        names: names,
        surnames: surnames,
        placeBorn: placeBorn,
        dateBorn: dateBorn,
        idbook: idbook
    };
    return object;
}


